llvm/utils/TableGen/DecoderEmitter.cpp

```diff
@@ -97,8 +97,8 @@ struct OperandInfo {
   bool HasCompleteDecoder;
   uint64_t InitValue;
 
-  OperandInfo(std::string D, bool HCD)
-      : Decoder(std::move(D)), HasCompleteDecoder(HCD), InitValue(0) {}
+  OperandInfo(const std::string &D, bool HCD)
```

Comment:
This can lead to extra copies depending on how this constructor is used. If it's called with a temporary, the original code would move the string to the argument and then move it again to the class. The new code would copy it to the argument.

Reply:
Thanks. I think this is covered here: https://abseil.io/tips/117  (the reason to use byval). I'll revert back to that.

Reply:
Done.